jetstream/external_config.py

```diff
@@ -39,6 +44,99 @@ def validate(self, experiment: jetstream.experimenter.Experiment) -> None:
         Analysis("no project", "no dataset", conf).validate()
 
 
+def validate_config_settings(config_file: Path) -> None:
+    """
+    Implemented to resolve a Github issue:
+        - https://github.com/mozilla/jetstream/issues/843
+
+    Loads external config file and runs a number of validation steps on it:
+    - Checks that all config keys/settings are lowercase
+    - Checks for missing core config keys
+    - Checks for unexpected core configuration keys
+    - Checks that all segments defined under experiment have configuration in segments section
+    - Checks if metric with custom config is defined in metrics.weekly or metrics.overall fields
+
+    Returns None, if issues found with the configuration an Exception is raised
+    """
+
+    config = toml.loads(config_file.read_text())
+
+    optional_core_config_keys = (
+        "metrics",
+        "experiment",
+        "segments",
+        "data_sources",
+        "friendly_name",
+        "description",
+    )  # TODO: double check that friendly_name and description are to be expected here
```

Comment:
yes, they are part of this list
```suggestion
    )
```